service/controller/v13/resource/s3object/desired.go

```diff
@@ -42,7 +42,7 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 		return output, microerror.Mask(err)
 	}
 
-	tlsAssets, err := r.encodeTLSAssets(ctx, certs, kmsKeyARN)
+	tlsAssets, err := r.encrypter.EncryptTLSAssets(ctx, customObject, certs)
```

Comment:
This is really cool that we work on this abstraction here now. The project improves that way quite nicely.